pkg/endpoint/endpoint.go

```diff
@@ -128,6 +128,10 @@ type Endpoint struct {
 	// recalculated on endpoint restore.
 	createdAt time.Time
 
+	//identityChangeTime store the time the endpoint security identity was changed.
+	// it is used for calculated endpoint progation delay.
+	identityChangeTime map[string]time.Time
```

Comment:
Could this be generalized into `lastUpdatedTime` which should account for all update events, not just identity changes.

Reply:
1. this condition is for create event :)
2. yes, it shoud be,  but current CEP type only contain identity info could be used to calculate this delay. i also want to know the method when don't change CEP CRD definition

Reply:
> this condition is for create event :)

Yes I agree, I guess I'm trying to argue that the propagation delay should also capture *updated* CEPs, not just newly created ones.

> yes, it shoud be, but current CEP type only contain identity info could be used to calculate this delay. i also want to know the method when don't change CEP CRD definition

The CEP CRD definition is not affected by the metrics or the introduction of CES, it is still https://github.com/cilium/cilium/blob/master/pkg/k8s/apis/cilium.io/v2/types.go#L45, aka any of these status field change could trigger a CEP update when CES is not enabled. Since this PR is trying to make the metrics more generic, I feel like it should account for all these other field changes. For this PR, I think it should be enough to just name the variables better (`identityChangeTime` -> `lastUpdatedTime`), then maybe in a follow up PR, we could introduce fields to the metrics to capture different types of update triggers. WDYT?

Reply:
:),thx! that's no problem! i will update after a while